File: articles/decision-science-in-crisis-management/python/run_all_crisis_management_workflows.py
#!/usr/bin/env python3
from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "crisis_risk_model.py",
    "escalation_trigger_model.py",
    "trust_and_communication_model.py",
    "crisis_response_comparison.py",
    "decision_record_exporter.py",
    "decision_science_crisis_management_simulation.py",
]

def main() -> None:
    for script in SCRIPTS:
        print(f"Running {script}...")
        subprocess.run([sys.executable, str(PYTHON_DIR / script)], check=True)
    print("All Python crisis-management workflows completed.")

if __name__ == "__main__":
    main()

File: articles/decision-science-in-crisis-management/r/run_all_crisis_management_workflows.R
args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)
script_path <- if (length(file_arg) > 0) normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE) else file.path(getwd(), "r", "run_all_crisis_management_workflows.R")
article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
r_dir <- file.path(article_root, "r")

scripts <- c(
  "decision_science_crisis_management_workflow.R",
  "crisis_response_profiles.R",
  "scenario_performance.R",
  "crisis_review_tables.R",
  "crisis_management_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R crisis-management workflows completed.")
